Amplify/Categories/DataStore/Model/Model.swift

```diff
@@ -46,6 +46,35 @@ extension Model {
         return identifier(schema: schema).stringValue
     }
 
+    public static func identifier(from id: String) -> ModelIdentifierProtocol {
+        return DefaultModelIdentifier<Self>.makeDefault(id: id)
+    }
+
+    public static func identifier(
+        of lazyReferenceField: ModelField,
+        from assoicatedModelName: String,
+        associatedFieldsData: [String: Persistable?]
+    ) -> ModelIdentifierProtocol? {
+        // according to codegen logic:
+        // https://github.com/aws-amplify/amplify-codegen/blob/main/packages/appsync-modelgen-plugin/src/utils/process-connections.ts#L41
+        // associated field name has format ${type}${field}${primaryKey} in camel case.
```

Comment:
can this move over to the datastore plugin package? At first glance, the reference to codegen/transformer is an implementation detail.

Looks only used in plugin's `Statement+Model` 
```
return DataStoreModelDecoder.lazyInit(
                    identifier: associatedFieldsData.flatMap {
                        modelType.identifier(of: field, from: modelName, associatedFieldsData: $0)
                    }
                )
```